core/src/main/java/com/google/errorprone/bugpatterns/CheckedExceptionNotThrown.java

```diff
@@ -64,8 +64,8 @@
 /** Flags checked exceptions which are claimed to be thrown, but are not. */
 @BugPattern(
     summary =
-        "This method cannot throw a checked exception that it claims to. This may cause consumers"
-            + " of the API to incorrectly attempt to handle, or propagate, this exception.",
+        "This method is overridable and cannot throw any of the checked exceptions it claims to. This may cause consumers"
```

Comment:
Shouldn't that say "final" rather than "overridable"?

Also, about the "any of the checked exceptions", is that really the current behavior? Shouldn't that be "one of the checked exceptions"? (in which case, the original phrasing was clearer; what I find unclear, as a non-native English speaker, is the "cannot", "will not" or "does not", or "never throws" would be clearer)

Reply:
Actually it should say "non-overridable". Final methods are non-overridable AND they also can't be hidden. Static methods are non-overridable but can be hidden. Private makes the method un-overridable through being inaccessible. I confirmed the check considers all these cases.

Yes you're right the original is clearer, but even clearer is to avoid the indefinite article, so "one or more".

Even as native English speaker we often miss points like this. To say "cannot" might suggest the method is trying to "throw new IOException" or the like. But this is about how the method throws/propagates unhandled exceptions. The method "will never" be required to throw the exception. "does not" is something to consider at runtime.